x-pack/plugins/maps/server/plugin.ts

```diff
@@ -147,27 +147,25 @@ export class MapsPlugin implements Plugin {
       return;
     }
 
-    let routesInitialized = false;
     let isEnterprisePlus = false;
+    let lastLicenseId: string | undefined;
     const emsSettings = new EMSSettings(mapsLegacyConfig, () => isEnterprisePlus);
     licensing.license$.subscribe((license: ILicense) => {
-      const basic = license.check(APP_ID, 'basic');
-
       const enterprise = license.check(APP_ID, 'enterprise');
       isEnterprisePlus = enterprise.state === 'valid';
-
-      if (basic.state === 'valid' && !routesInitialized) {
-        routesInitialized = true;
-        initRoutes(
-          core.http.createRouter(),
-          license.uid,
-          emsSettings,
-          this.kibanaVersion,
-          this._logger
-        );
-      }
+      lastLicenseId = license.uid;
     });
 
+    initRoutes(
+      core.http.createRouter(),
+      () => {
+        return lastLicenseId;
+      },
```

Comment:
Not critical at all (i.e.- feel free to ignore), but easy one-liner: `() => lastLicenseId`